polar-core/src/filter.rs

```diff
@@ -541,9 +562,90 @@ where
     std::iter::once(x).collect()
 }
 
+fn normalize(t: Term) -> Vec<Term> {
```

Comment:
A comment summing up the normalization that this applies would be helpful.

Reply:
Can this use the code in `normalize` instead?

Reply:
Is this fixing something or changing the disjunctive normal form?

Reply:
It's just flattening the tree into a vector, so the name isn't really right, I'll change it to `vec_of_ands` or something

Reply:
Ohhhh I totally missed that